matchtree.go

```diff
@@ -983,14 +991,20 @@ func (d *indexData) newMatchTree(q query.Q, opt matchTreeOpt) (matchTree, error)
 		optCopy := opt
 		optCopy.DisableWordMatchOptimization = true
 
-		subMT, err := d.newMatchTree(s.Expr, optCopy)
+		expr, wasAnchored := s.Expr, false
+		if regexpExpr, ok := expr.(*query.Regexp); ok {
+			expr, wasAnchored = stripAnchors(regexpExpr)
+		}
+
+		subMT, err := d.newMatchTree(expr, optCopy)
 		if err != nil {
 			return nil, err
 		}
 
 		if substr, ok := subMT.(*substrMatchTree); ok {
 			return &symbolSubstrMatchTree{
 				substrMatchTree: substr,
+				exact:           wasAnchored,
```

Comment:
one last bug here, I think the regexp we pass to symbolRegexpMatchTree (if we get to that path) will now exclude anchors.